lodash.js

```diff
@@ -5440,14 +5441,23 @@
       if (length && !values && !isArray(props[0])) {
         values = [];
       }
+
       while (++index < length) {
-        var key = props[index];
-        if (values) {
-          result[key] = values[index];
-        } else if (key) {
-          result[key[0]] = key[1];
+        var key = props[index],
+            resultProp,
+            resultValue;
+
+        resultProp = values ? key: key[0];
+        resultValue = values ? values[index]: key[1];
+
+        if (result[resultProp]) {
+          result[resultProp] = Array.isArray(result[resultProp]) ? result[resultProp]: [result[resultProp]];
```

Comment:
This won't be entirely sufficient as

`_.zipObject([['test', [1]]]` => `{test: [[1]]}`